{
  try {
    Assert.assertEquals(this.expectedAuthenticate,this.handler.authenticate(this.credential));
  }
 catch (  Exception e) {
    if (this.handler.supports(this.credential)) {
      e.printStackTrace();
      Assert.fail("Unexpected authentication error: " + e);
    }
  }
}
